### ISSUE TITLE
no-return-assign: diagnostic anchored on the inner assignment instead of the ReturnStatement, breaking eslint-disable-next-line on multi-line returns

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25628]
## Summary

With `no-return-assign` set to `["error", "always"]`, oxlint anchors the diagnostic on the inner `AssignmentExpression` instead of the `ReturnStatement` (as ESLint does). For multi-line `return (...)` statements the assignment lives on a different line than the `return` keyword, so a `// eslint-disable-next-line no-return-assign` placed above the `return` fails to suppress the finding — and is additionally reported as an unused directive under `--report-unused-disable-directives`.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0.

## Reproduction

`a.js`:

```js
const cache = {}
const o = {
	get x() {
		// eslint-disable-next-line no-return-assign
		return (
			cache.x ??
			(cache.x = build())
		)
	},
}
```

`b.js` — same body without the directive (`return` on line 4), to compare the report span:

```js
const cache = {}
const o = {
	get x() {
		return (
			cache.x ??
			(cache.x = build())
		)
	},
}
```

`.oxlintrc.json`:

```json
{"plugins": [], "categories": {"correctness": "off"}, "rules": {"no-return-assign": ["error", "always"]}}
```

Run:

```
oxlint -c .oxlintrc.json --report-unused-disable-directives a.js b.js
```

## Expected

ESLint anchors the report on the `ReturnStatement`. On `b.js`, ESLint 7.23.0 and ESLint 9.39.4 (flat config, core rule) both report `{"line":4,"column":3,"endLine":7,"endColumn":4,"nodeType":"ReturnStatement"}`. Consequently on `a.js` the `// eslint-disable-next-line no-return-assign` on line 4 suppresses the finding, and `--report-unused-disable-directives` reports nothing (directive live, zero errors).

## Actual (oxlint 1.78.0)

```
a.js:7:5: error eslint(no-return-assign): Return statements should not contain an assignment. help: Did you mean to use `==` instead of `=`?
b.js:6:5: error eslint(no-return-assign): Return statements should not contain an assignment. help: Did you mean to use `==` instead of `=`?
a.js:4:3: warning: Unused eslint-disable directive (no problems were reported).
```

The diagnostic is anchored on the inner assignment (`a.js:7:5`), so the directive on line 4 does not match: the error leaks through **and** the directive is flagged unused. A single-line `return cache.x ?? (cache.x = build())` is suppressed correctly — only because the assignment happens to share the `return`'s line — confirming the bug is purely the diagnostic location/node, not detection.

## Cross-checked

| Tool | Version | Anchor on `b.js` | Directive above multi-line `return` |
| --- | --- | --- | --- |
| ESLint (core) | 7.23.0 | `ReturnStatement` (4:3–7:4) | suppresses; no unused-directive warning |
| ESLint (core, flat config) | 9.39.4 | `ReturnStatement` (4:3–7:4) | suppresses; no unused-directive warning |
| oxlint | 1.78.0 | inner assignment (6:5) | fails to suppress; unused-directive warning |

Suggested fix: anchor `no-return-assign` reports on the `ReturnStatement` / `ArrowFunctionExpression` (the node ESLint reports), not the nested assignment, so that `eslint-disable-next-line` directives placed above the `return` match.
